### ISSUE TITLE
In progress: UI broken in 2022.3+

### ISSUE THREAD

**Issue by PiotrMachowski** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#365]
### Checklist

- [X] I have updated the card to the latest version available
- [X] I have cleared the cache of my browser
- [X] I have checked if the problem is already reported

### The problem

Modes drop-down broken since 2022.3

### What version of a card has described problem?

v2.0.10

### What vacuum model do you have problems with?

Any

### Which integration do you use to control your vacuum (link)?

https://www.home-assistant.io/integrations/xiaomi_miio/

### What browser (browsers/apps) does have this problem?

Any

### What version of Home Assistant do you use?

2022.3.0

### What type of installation are you running?

Home Assistant OS

### Card's configuration

```yaml
N/A
```

**Comment by LeandroIssa** [comment]
The card has appeared like this to me. I've already cleared the cache, I'm on card + recent version and Home Assistant version 2022.3.

![screencapture-192-168-31-206-8123-lovelace-teste-2022-03-02-20_40_36](https://user-images.githubusercontent.com/67451572/156468307-ecaf7f4d-ad16-4b6b-9101-0c421b385111.png)

**Comment by lfrlucas** [comment]
I confirm this issue after upgrading Home Assistant to 2022.3

**Comment by PiotrMachowski** [comment]
@variada yup, that's the cause. I will try to fix it today

**Comment by jscherry** [comment]
Same problem

**Comment by escafoide** [comment]
Hi, any idea when a new version will be released with this issue resolved? Thank you

**Comment by PiotrMachowski** [comment]
@escafoide it will be released when it is ready

**Comment by mateuszdrab** [comment]
@PiotrMachowski Thanks for looking into it Piotr.
Just saw it's broken on my laptop but looks fine on mobile.

**Comment by YOZZOPANOZZO** [comment]
@PiotrMachowski Thanks!

**Comment by bravesfan171** [comment]
@PiotrMachowski Thanks for getting on this issue!

**Comment by sasukebinbin** [comment]
@PiotrMachowski  Thanks for resolve the issue

**Comment by sagitt** [comment]
thank you

**Comment by alesoft73** [comment]
The best thanks are those made by offering a beer ...
1 beer fo you;)

**Comment by digitalit** [comment]
@PiotrMachowski 
I'm no expert on github issues but i've been told in other places that one should not write like "don't work for me either or following"

I guess this will only stress the developer with unnecessary notices. What I’ve been told it’s better to just subscribe to the issue.

Anyway, I didn’t buy you a beer but I bought you a coffee that has proven my coding experience better than beer :)

Best regards
Peter Sweden

**Comment by PiotrMachowski** [comment]
@digitalit well, these comments are a bit unnecessary, because it won't work for all 2022.03+ installations 😉 

There is a lot of pressure on this issue, but I don't have much time lately, so progress is quite slow 😕

**Comment by mateuszdrab** [comment]
I didn't mean to start a wave of thanks as a pressure method.
I don't mind waiting for the fix - it's not a critical issue being a UI glitch and I don't use that component very often.
It definitely is a little cherry on the cake on my setup though - love the map integration.
But yeah, I've materialised the thanks into some beers.

**Comment by PiotrMachowski** [comment]
I'm really fond of this card as well, it took a lot of work, but the result is quite satisfying 🙂 personally I really like its universality - it probably supports every vacuum that works with HA (some might require a bit of tinkering in config though)

**Comment by pannal** [comment]
> I didn't mean to start a wave of thanks as a pressure method. I don't mind waiting for the fix - it's not a critical issue being a UI glitch and I don't use that component very often. It definitely is a little cherry on the cake on my setup though - love the map integration. But yeah, I've materialised the thanks into some beers.

Yep, this is plugin great. The breaking changes were expectable, but I won't complain. It's still usable enough :)

**Comment by Adelich27** [comment]
where solution?

**Comment by pannal** [comment]
> where solution?

Where brain?

He's working on it, as he said above.

**Comment by digitalit** [comment]
@Adelich27 , thats just rude :(

**Comment by Adelich27** [comment]
> @Adelich27 , that's just rude :(

I'm sorry, I'm just from Russia and I don't understand you well. Now I understand everything, thank you very much.

**Comment by vietlehung** [comment]
Hi, and thank you for hard work. UI broken also happen to our old ver. 1.x. that bring me here.
For some reasons I still maintain the 1.x till date:
- I have Xiaomi vaccum V.1 then it's very basic one only work with send command, With Map card virtually I can organize cleanning sequence between rooms.
- We can bypass limit of 5 zones selections.
- UI is simple but that just focus on the most use features. I can add  "pause/play" and "stop" button next to maps card then it work fine until HA update 2022.3.

I tried to fix problem of the UI by updating to our latest V.2.10, then I can adjust max selections value to more than 5 zones and acttualy we can select more than 5 zones but the robot stop itself after finishing the 1st 5 zones. 
Could you take a look into our old V.1.x UI and 5 zones limit of the latest one. 
Best regards

**Comment by PiotrMachowski** [comment]
@vietlehung 
* It's possible to use send_command preset in 2.x, it will use the same service calls as 1.x, cleaning sequence is kept as well
* It's already possible using `max_selections` property, limitation to 5 zones is on vacuum's side (the same problem as in 1.x - can be bypassed using a script)
* It's possible to hide unwanted elements of the card (e.g. tiles), see #162,

I don't really want to reanimate old version of the card, I suggest migrating to new version and adjusting it to your needs

**Comment by vietlehung** [comment]
Hi @PiotrMachowski , thank you very much for your guidance. The UI is now more neat, I'm trying now to learn how can we call a script with V2.x. Here is my old config in V1.x, we only need to call service directly with a script 

/////
calibration_points:
default_mode: predefined_zones
entity: vacuum.mi_robot
map_image: /local/custom_lovelace/xiaomi_vacuum_map_card/map2.png
service: script.vacuum_send_command_multiple_zones
ignore_zones_limit: true
type: custom:xiaomi-vacuum-map-card
zones:
 - ....
 //////

I found that we have an example of script call "vacuum_follow_path" however it mentioned as template supported level. I tried to place this script service call ("service: script.vacuum_send_command_multiple_zones") in some presets but not successful bypass 5 zones limit. So it failed to replace default service currenlty. Should I need to make new template for this service call?
Best regards

**Comment by PiotrMachowski** [comment]
Here you have a config that works identically to to v1.x:
```yaml
map_modes:
  - template: vacuum_clean_zone
    max_selections: 20
    service_call_schema:
      service: script.vacuum_send_command_multiple_zones
      service_data:
        entity_id: "[[entity_id]]"
        command: app_zoned_clean
        params: "[[selection]]"
  - template: vacuum_goto
    service_call_schema:
      service: script.vacuum_send_command_multiple_zones
      service_data:
        entity_id: "[[entity_id]]"
        command: app_goto_target
        params: "[[selection]]"
```

**Comment by vietlehung** [comment]
Hi @PiotrMachowski , deeply thanks for your help. it well works now.

**Comment by jimz011** [comment]
Was coming here as I had an issue as well, but mentioning that would be redundant.

It's good enough to know that there is being worked on even if development isn't fast. As someone that creates experiences for Home Assistant myself I totally understand and you should definitely not feel pressured! If people want to move on they'll move on, but I highly doubt there is anything much better than this card for vacuums.

So...... Take ALL the time you need!

**Comment by trixor** [comment]
Thank you @PiotrMachowski for working through this! I'm new to Home Assistant in general, and I must say I'm very impressed with this card and your work! Perhaps one day I'll be able to contribute code myself. Sending some beers as a thank you!

**Comment by qilza** [comment]
Hello, I can't  understand where is solutions for UI isuue. Please share full resolution 
this is my card

type: custom:xiaomi-vacuum-map-card
map_locked: true
map_source:
  camera: camera.xiaomi_cloud_map_extractor
calibration_source:
  camera: true
entity: vacuum.roborock
vacuum_platform: default
preset_name: Первый Этаж
map_modes:
  - name: Выбор зоны уборки
    icon: mdi:select-drag
    selection_type: MANUAL_RECTANGLE
    max_selections: 2
    repeats_type: INTERNAL
    max_repeats: 2
    service_call_schema:
      service: vacuum.do_zones
      service_data:
        zones: '[[selection]]'
        repeats: '[[repeats]]'
        predefined: 'false'
      target:
        entity_id: '[[entity_id]]'
  - name: Помещения
    icon: mdi:floor-plan
    selection_type: PREDEFINED_RECTANGLE
    max_selections: 5
    repeats_type: INTERNAL
    max_repeats: 5
    service_call_schema:
      service: vacuum.do_zones
      service_data:
        zones: '[[selection]]'
        repeats: '[[repeats]]'
        predefined: 'true'
      target:
        entity_id: '[[entity_id]]'
    predefined_selections:
      - zones:
          - - 28080
            - 33720
            - 17800
            - 29180
        icon:
          name: mdi:vacuum
          x: 22932
          'y': 31539
      - zones:
          - - 21390
            - 28870
            - 17600
            - 24400
        icon:
          name: mdi:vacuum
          x: 19500
          'y': 26800
      - zones:
          - - 21390
            - 28870
            - 28500
            - 24400
        icon:
          name: mdi:vacuum
          x: 25900
          'y': 27800
title: Готово
additional_presets:
  - preset_name: Второй Этаж
    entity: vacuum.roborock
    map_source:
      camera: camera.xiaomi_cloud_map_extractor
    calibration_source:
      camera: true
    map_modes:
      - name: Выбор зоны уборки
        icon: mdi:select-drag
        selection_type: MANUAL_RECTANGLE
        max_selections: 2
        repeats_type: INTERNAL
        max_repeats: 2
        service_call_schema:
          service: vacuum.do_zones
          service_data:
            zones: '[[selection]]'
            repeats: '[[repeats]]'
            predefined: 'false'
          target:
            entity_id: '[[entity_id]]'
      - name: Помещения
        icon: mdi:mdi:floor-plan
        selection_type: PREDEFINED_RECTANGLE
        max_selections: 5
        repeats_type: INTERNAL
        max_repeats: 5
        service_call_schema:
          service: vacuum.do_zones
          service_data:
            zones: '[[selection]]'
            repeats: '[[repeats]]'
            predefined: 'true'
          target:
            entity_id: '[[entity_id]]'
        predefined_selections:
          - zones:
              - - 28080
                - 33720
                - 17800
                - 29180
            icon:
              name: mdi:vacuum
              x: 22932
              'y': 31539
          - zones:
              - - 21390
                - 28870
                - 17600
                - 24400
            icon:
              name: mdi:vacuum
              x: 19500
              'y': 26800
          - zones:
              - - 21390
                - 28870
                - 28500
                - 24400
            icon:
              name: mdi:vacuum
              x: 25900
              'y': 27800

**Comment by qilza** [comment]
![image](https://user-images.githubusercontent.com/99974060/162424706-7e1bf4ec-b356-443e-97d7-9b52828a25f6.png)

**Comment by pannal** [comment]
> Hello, I can't understand where is solutions for UI isuue. Please share full resolution this is my card
> 
> type: custom:xiaomi-vacuum-map-card map_locked: true map_source: camera: camera.xiaomi_cloud_map_extractor calibration_source: camera: true entity: vacuum.roborock vacuum_platform: default preset_name: Первый Этаж map_modes:
> 
> * name: Выбор зоны уборки
>   icon: mdi:select-drag
>   selection_type: MANUAL_RECTANGLE
>   max_selections: 2
>   repeats_type: INTERNAL
>   max_repeats: 2
>   service_call_schema:
>   service: vacuum.do_zones
>   service_data:
>   zones: '[[selection]]'
>   repeats: '[[repeats]]'
>   predefined: 'false'
>   target:
>   entity_id: '[[entity_id]]'
> * name: Помещения
>   icon: mdi:floor-plan
>   selection_type: PREDEFINED_RECTANGLE
>   max_selections: 5
>   repeats_type: INTERNAL
>   max_repeats: 5
>   service_call_schema:
>   service: vacuum.do_zones
>   service_data:
>   zones: '[[selection]]'
>   repeats: '[[repeats]]'
>   predefined: 'true'
>   target:
>   entity_id: '[[entity_id]]'
>   predefined_selections:
>   
>   * zones:
>     
>     * * 28080
>       * 33720
>       * 17800
>       * 29180
>         icon:
>         name: mdi:vacuum
>         x: 22932
>         'y': 31539
>   * zones:
>     
>     * * 21390
>       * 28870
>       * 17600
>       * 24400
>         icon:
>         name: mdi:vacuum
>         x: 19500
>         'y': 26800
>   * zones:
>     
>     * * 21390
>       * 28870
>       * 28500
>       * 24400
>         icon:
>         name: mdi:vacuum
>         x: 25900
>         'y': 27800
>         title: Готово
>         additional_presets:
> * preset_name: Второй Этаж
>   entity: vacuum.roborock
>   map_source:
>   camera: camera.xiaomi_cloud_map_extractor
>   calibration_source:
>   camera: true
>   map_modes:
>   
>   * name: Выбор зоны уборки
>     icon: mdi:select-drag
>     selection_type: MANUAL_RECTANGLE
>     max_selections: 2
>     repeats_type: INTERNAL
>     max_repeats: 2
>     service_call_schema:
>     service: vacuum.do_zones
>     service_data:
>     zones: '[[selection]]'
>     repeats: '[[repeats]]'
>     predefined: 'false'
>     target:
>     entity_id: '[[entity_id]]'
>   * name: Помещения
>     icon: mdi:mdi:floor-plan
>     selection_type: PREDEFINED_RECTANGLE
>     max_selections: 5
>     repeats_type: INTERNAL
>     max_repeats: 5
>     service_call_schema:
>     service: vacuum.do_zones
>     service_data:
>     zones: '[[selection]]'
>     repeats: '[[repeats]]'
>     predefined: 'true'
>     target:
>     entity_id: '[[entity_id]]'
>     predefined_selections:
>     
>     * zones:
>       
>       * * 28080
>         * 33720
>         * 17800
>         * 29180
>           icon:
>           name: mdi:vacuum
>           x: 22932
>           'y': 31539
>     * zones:
>       
>       * * 21390
>         * 28870
>         * 17600
>         * 24400
>           icon:
>           name: mdi:vacuum
>           x: 19500
>           'y': 26800
>     * zones:
>       
>       * * 21390
>         * 28870
>         * 28500
>         * 24400
>           icon:
>           name: mdi:vacuum
>           x: 25900
>           'y': 27800

There is none yet.

@PiotrMachowski maybe add "UNRESOLVED" to the issue title; this gets polluted quite heavily by people who can't or won't backtrack the comments.

**Comment by jimz011** [comment]
It is an open issue, ANY open issue is basically unresolved. Would seem redundant to me and honestly people should stop being so lazy and just read. The developer puts months of work in cards like these, but people that use them are even too lazy to give it one minute of reading time?

**Comment by pannal** [comment]
> It is an open issue, ANY open issue is basically unresolved. Would seem redundant to me and honestly people should stop being so lazy and just read. The developer puts months of work in cards like these, but people that use them are even too lazy to give it one minute of reading time?

Well yeah. But this is not an ideal world, people are people, and I'd bet most of the people coming here never even have used Github in the past and just relied on being able to click update in HACS.

**Comment by qilza** [comment]
Guys, i have readed all but didn't understand. that why i have wtired comments. Sorry for mis understanding. Thx a lot for your support

**Comment by jimz011** [comment]
@pannal fair point.

**Comment by mateuszdrab** [comment]
I think it's also due to the language barrier. Though, that's where Google Translate comes in.

**Comment by SeminolesFan78** [comment]
So glad I found this issue. Does the title mean that rolling HA back to 2022.2.9 would make this card work as expected? Has anyone tried that yet and can report success?

**Comment by jimz011** [comment]
@SeminolesFan78 I haven't tried that, but I am pretty sure that that would indeed work.

**Comment by MajorWedgie** [comment]
To anyone who has updated to 2022.3+ , isn't it only the drop down menus that doesn't work? 
You can still get zone coordinates with debug set to true, right? Or no?

**Comment by werra11** [comment]
> To anyone who has updated to 2022.3+ , isn't it only the drop down menus that doesn't work? You can still get zone coordinates with debug set to true, right? Or no?

Correct, everything works fine. Only UI is bugged.

**Comment by garv3** [comment]
Still no sulution yet? 😢

**Comment by digitalit** [comment]
![vacum](https://user-images.githubusercontent.com/6569136/166101414-090b9f91-3dfa-4b2c-b3c5-22c42149a9f8.png) 

I might be stupid but I had honestly forgotten about this bug.

Sure i have two opened dropdowns but eveerything works fine, maybe i'm missing a serious problem but for me it works fine.

My biggest fear is that @PiotrMachowski will trash this whole project due to all the nagging

**Comment by jimz011** [comment]
@digitalit then don't mention the developer. He will get a notification that you have mentioned him. But your comment doesn't help the project. So if I were the dev of this project, I would consider people mentioning me (without any good reason) nagging. Simply because I would get notified for nothing.

**Comment by PiotrMachowski** [comment]
@jimz011 I receive notifications about every comment anyway, but it doesn't bother me so much 😉  It's much better than issues created by people that haven't read the documentation 😃

**Comment by jerome6994** [comment]
Hello
I'm sorry I looked at the previous comments but I don't have a solution to this problem or is it the last update that broke this?
I can't say if it worked before because I had a calibration problem that prevented me from seeing this.
sorry once again I am also new to Home Assistant
I am open to your solutions if they exist or else I will be patient

![image](https://user-images.githubusercontent.com/66921143/167086829-7a224aa9-255a-4331-b6d0-b716fa9097e4.png)

**Comment by PiotrMachowski** [comment]
@jerome6994 Unfortunately I haven't made a fix so far :(

**Comment by jerome6994** [comment]
Ok I thought I missed something.
No worries and no stress it's already a job at the top BRAVO

**Comment by s-hutter** [comment]
To clean this up a little until there's a proper fix available, you can install [Card Mod](https://github.com/thomasloven/lovelace-card-mod#card-mod-3) and then add this bit of css code to the end of your Vacuum Map Card in the YAML code editor:
 ```
style: |
  .modes-dropdown-menu > .modes-dropdown-menu-listbox {display: none;}
  .modes-dropdown-menu:hover > .modes-dropdown-menu-listbox {display: block;}
  .modes-dropdown-menu:hover > .modes-dropdown-menu-button {display: none;}
  .modes-dropdown-menu-entry-button-wrapper.first {border-start-start-radius: var(--map-card-internal-big-radius);}
  .modes-dropdown-menu-entry-button-wrapper.last {border-end-start-radius: var(--map-card-internal-big-radius);}
  .map-actions-list {height: 50px;}
```

**Comment by mrwogu** [comment]
I have prepared a fix for this problem: #392

**Comment by PiotrMachowski** [comment]
Well, this fix fixes dropdown functionality, but breaks card design, so unfortunately I can't merge it

**Comment by jrlacharnay** [comment]
> To clean this up a little until there's a proper fix available, you can install [Card Mod](https://github.com/thomasloven/lovelace-card-mod#card-mod-3) and then add this bit of css code to the end of your Vacuum Map Card in the YAML code editor:
> 
> ```
> style: |
>  .modes-dropdown-menu > .modes-dropdown-menu-listbox {display: none;}
>  .modes-dropdown-menu:hover > .modes-dropdown-menu-listbox {display: block;}
>  .modes-dropdown-menu:hover > .modes-dropdown-menu-button {display: none;}
>  .modes-dropdown-menu-entry-button-wrapper.first {border-start-start-radius: var(--map-card-internal-big-radius);}
>  .modes-dropdown-menu-entry-button-wrapper.last {border-end-start-radius: var(--map-card-internal-big-radius);}
>  .map-actions-list {height: 50px;}
> ```

Thank you @s-hutter this works perfectly!

**Comment by mateuszdrab** [comment]
> > To clean this up a little until there's a proper fix available, you can install [Card Mod](https://github.com/thomasloven/lovelace-card-mod#card-mod-3) and then add this bit of css code to the end of your Vacuum Map Card in the YAML code editor:
> > 
> > ```
> > style: |
> >  .modes-dropdown-menu > .modes-dropdown-menu-listbox {display: none;}
> >  .modes-dropdown-menu:hover > .modes-dropdown-menu-listbox {display: block;}
> >  .modes-dropdown-menu:hover > .modes-dropdown-menu-button {display: none;}
> >  .modes-dropdown-menu-entry-button-wrapper.first {border-start-start-radius: var(--map-card-internal-big-radius);}
> >  .modes-dropdown-menu-entry-button-wrapper.last {border-end-start-radius: var(--map-card-internal-big-radius);}
> >  .map-actions-list {height: 50px;}
> > ```
> 
> Thank you @s-hutter this works perfectly!

Nice little tweak, but it works, I mean when the menu expands it won't collapse but it's fine.

**Comment by befantasy** [comment]
Hi, PiotrMachowski. The drop down menus have broke down for a long time and actually the Map modes options are not so important for some people like me. Could you consider providing an option to disable these drop down menus until the issue is solved?

**Comment by PiotrMachowski** [comment]
@befantasy Issue should be solved today ;)

**Comment by PiotrMachowski** [comment]
Fixed in v2.0.11

**Comment by jambuu** [comment]
Hi @PiotrMachowski, thanks for fixing this. I have installed it with hacs, restarted, reload resources but the problem still persists. Did I do something wrong? 
![Screenshot_20220509_210601_io homeassistant companion android](https://user-images.githubusercontent.com/20259378/167480931-0e6a34db-f137-4697-ae77-0f8d23c94173.jpg)

**Comment by PiotrMachowski** [comment]
@jambuu is it broken in incognito mode as well?

**Comment by rufinus** [comment]
fix is working.
![image](https://user-images.githubusercontent.com/343160/167481381-13bcc52b-edea-4af8-999c-804aa782cfec.png)


Thank you.

**Comment by jambuu** [comment]
Hi @PiotrMachowski, I was accessing through the home assistant app in android and that I was able to get it working. Probably still caching something somewhere.. But on browser in incognito, that worked great! Thank you!

**Comment by heso** [comment]
after reloading the cache everything is ok, ty

**Comment by jrlacharnay** [comment]
Hi,

In my config I disabled the icons & tiles :

```
icons: []
tiles: []

```

And now since the very last update, the menu is sometimes expanding down and is "cut" by the card :

![image](https://user-images.githubusercontent.com/20613054/167508643-99942831-263b-48f4-b820-b2db1eb74928.png)

It works well when it expands up. It depends its position in the screen.

Should I open a new issue?

Thanks!

**Comment by PiotrMachowski** [comment]
@jrlacharnay ugh... 😞 Yeah, please create a separate issue

**Comment by befantasy** [comment]
it works！thanks a lot！

**Comment by jerome6994** [comment]
thank you very much for the work the fix corrects the concerns that I raised a few days ago.

**Comment by Xitro01** [comment]
Hi Piotr, I still have some issues with the UI. When you put the map in a pop-up it shifts the map upwards when selecting the dropdown list.

After clicking something in this list it closes the pop-up. It never had these issues before the 2022.3 update.
![x](https://user-images.githubusercontent.com/19844787/167604456-978726ca-423e-4aee-ace1-30855a122c12.PNG)

**Comment by PiotrMachowski** [comment]
@Xitro01 can you share your config?

**Comment by Xitro01** [comment]
```
  - platform: xiaomi_cloud_map_extractor
    host: -
    token: !secret xiaomi_miio
    username: -
    password: !secret password_xiaomi
    name: vacuum_map
    country: de
    colors:
      color_robo: [224, 224, 224]
      color_charger: [27, 117, 27]
      color_path: [30, 80, 113]
      color_goto_path: [30, 80, 113]
      color_map_wall_v2: [30, 30, 30]
      color_obstacle: [30, 30, 30]
      color_virtual_walls: [255, 0, 0, 60]
      color_no_go_zones_outline: [255, 0, 0, 60]
      color_no_go_zones: [255, 0, 0, 28]
      color_scan: [0, 0, 0, 0]
      color_map_inside: [0, 0, 0, 0]
      color_map_outside: [0, 0, 0, 0]
      color_grey_wall: [0, 0, 0, 0]
      color_map_wall: [0, 0, 0, 0]
    room_colors:
      1: [42, 46, 48]
      2: [42, 46, 48]
      16: [42, 46, 48]
      17: [42, 46, 48]
      18: [42, 46, 48]
    draw:
      - all
    map_transformation:
      scale: 2
      trim:
        top: 32
        right: 21
        bottom: 25
        left: 18
    sizes:
      charger_radius: 7
      vacuum_radius: 7
    attributes:
      - calibration_points
```

**Comment by Xitro01** [comment]
```
action: fire-dom-event
browser_mod:
  command: popup
  title: Robot Stofzuiger
  style:
    .: |
      :host .content {
        width: calc(385px + 510px);
        max-width: 90vw;
      }
      /* phone */
      @media screen and (max-width: 800px) {
        :host .content {
          max-width: 100vw;
        }
      }
    layout-card$grid-layout:
      $: |
        hui-vertical-stack-card {
          border-right: 1.5px solid rgba(0, 0, 0, 0.2);
          border-radius: 0;
          transition: none;
        }
        /* phone */
        @media screen and (max-width: 800px) {
          hui-vertical-stack-card {
            border-right: none;
            border-bottom: 1.5px solid rgba(0, 0, 0, 0.2);
          }
        }
      $hui-vertical-stack-card:
        $: |
          hui-horizontal-stack-card {
            padding: 0em 2em 2.3em 2em;
          }
        $hui-entities-card$: |
          .card-content {
            padding: var(--card-content-padding);
          }
        $hui-horizontal-stack-card$: |
          #root {
            justify-content: space-evenly;
            max-width: 82vw; /* iphonex */
          }
      $xiaomi-vacuum-map-card:
        $: |
          div.map-wrapper, div.controls-wrapper {
            padding: var(--card-content-padding);
            padding-top: 0;
            padding-bottom: 0;
          }
          div.tiles-wrapper, div.vacuum-controls, #map-zoomer-overlay {
            display: none;
          }
          div.map-controls-wrapper {
            padding-top: 18px;
          }
          paper-button {
            --mdc-icon-size: 2em;
            color: #9da0a2 !important;
            padding: 0.42em;
          }
          ha-card {
            --map-card-primary-color: #FFFFFF10;
            --map-card-secondary-color: none;
            --map-card-secondary-text-color: #9da0a2;
            --map-card-zoomer-background: none;
            --map-card-predefined-rectangle-fill-color: none;
            --map-card-predefined-rectangle-line-color: none;
            --map-card-predefined-rectangle-line-color-selected: rgba(255, 255, 255, 0.5);
            --map-card-predefined-rectangle-fill-color-selected: rgba(42, 124, 194, 0.1);
            --map-card-predefined-rectangle-icon-background-color-selected: rgb(42, 124, 194);
            --map-card-internal-big-radius: 0.6em !important;
          }
  card:
    type: custom:layout-card
    layout_type: custom:grid-layout
    layout:
      grid-template-columns: 385px 510px
      grid-template-rows: 1fr
      grid-template-areas: |
        "info map"
      mediaquery:
        #phone
        "(max-width: 800px)":
          grid-template-columns: 1fr
          grid-template-rows: repeat(2, 1fr)
          grid-template-areas: |
            "info"
            "map"
    cards:
      #################################################
      #                                               #
      #                 Settings                      #
      #                                               #
      #################################################

      - type: vertical-stack
        view_layout:
          grid-area: info
        cards:
          - type: entities
            title: Instellingen
            card_mod:
              class: header
            entities:
              - entity: vacuum.roborock_vacuum_s5

              - entity: sensor.roborock_vacuum_s5_last_clean_end

              - type: custom:bar-card
                width: 55%
                height: 2em
                decimal: 0
                unit_of_measurement: "%"
                positions:
                  icon: outside
                  indicator: "off"
                  name: outside
                severity:
                  - color: "#303435"
                    from: 11
                    to: 100
                  - color: "#6d2525"
                    from: 0
                    to: 10
                entity_row: true
                entities:
                  - entity: vacuum.roborock_vacuum_s5
                    attribute: battery_level
                    name: Batterij
                    icon: mdi:battery

                  - entity: sensor.template_vacuum_filter

                  - entity: sensor.template_vacuum_mainbrush

                  - entity: sensor.template_vacuum_sidebrush

                  - entity: sensor.template_vacuum_sensors

              - entity: select.vacuum_speed
                name: Stofzuiger Snelheid

          - type: horizontal-stack
            cards:
              - type: custom:button-card
                entity: vacuum.roborock_vacuum_s5
                icon: mdi:play-pause
                tap_action:
                  action: call-service
                  service: >
                    [[[
                      return entity.state === 'docked' || entity.state === 'paused'
                        ? 'vacuum.start'
                        : 'vacuum.pause';
                    ]]]
                  service_data:
                    entity_id: >
                      [[[ return entity.entity_id; ]]]
                template: icon_only

              - type: custom:button-card
                entity: vacuum.roborock_vacuum_s5
                icon: mdi:battery-charging
                tap_action:
                  action: call-service
                  service: vacuum.return_to_base
                  service_data:
                    entity_id: >
                      [[[ return entity.entity_id; ]]]
                template: icon_only

              - type: custom:button-card
                icon: mdi:delete-empty
                tap_action:
                  action: call-service
                  service: vacuum.send_command
                  service_data:
                    entity_id: vacuum.roborock_vacuum_s5
                    command: app_goto_target
                    params: [25954, 25539]
                template: icon_only

      #################################################
      #                                               #
      #                 MAP                           #
      #                                               #
      #################################################

      - type: custom:xiaomi-vacuum-map-card
        title: Kaart
        view_layout:
          grid-area: map
        card_mod:
          class: header
        language: nl
        entity: vacuum.roborock_vacuum_s5
        map_source:
          camera: camera.vacuum_map
        calibration_source:
          camera: true
        map_locked: true
        map_modes:
          - template: vacuum_clean_zone
          - template: vacuum_clean_zone_predefined
            predefined_selections:
              - zones: [[26194, 27632, 29072, 31636]] #Eetkamer
                icon:
                  name: mdi:silverware-fork-knife
                  x: 27630
                  y: 29315
              - zones: [[24024, 23761, 25179, 29041]] #Gang
                icon:
                  name: mdi:door
                  x: 24567
                  y: 26337
              - zones: [[25110, 23973, 29049, 28946]] #Woonkamer
                icon:
                  name: mdi:sofa
                  x: 27183
                  y: 26625
              - zones: [[24241, 29779, 26291, 33854]] #Keuken
                icon:
                  name: mdi:pot-steam
                  x: 25273
                  y: 31413
          - template: vacuum_goto
          - template: vacuum_follow_path

```

**Comment by jrlacharnay** [comment]
> @jrlacharnay ugh... 😞 Yeah, please create a separate issue

https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/397

Thanks a lot for your support!

**Comment by zollek** [comment]
> Fixed in v2.0.11

@PiotrMachowski Thanks for the update.
I can confirm it's working.

**Comment by rensenti** [comment]
Thanks for the update, unfortunately I'm experiencing the same issue as jrlacharnay. The popup disappears when selecting a different map mode.

```yaml
popup_cards:
  vacuum.roborocks7:
    title: Usisavac
    card:
      type: vertical-stack
      cards:    
      - type: custom:xiaomi-vacuum-map-card
        icons: []
        tiles:
          - label: Voda
            entity: sensor.roborock_propusnost_vode
            icon: mdi:water
          - label: Brisanje
            entity: sensor.roborock_brisanje
            icon: mdi:wiper
          - label: Usisavanje ili brisanje
            entity: sensor.roborock_usisavanje
            icon: mdi:vacuum
        preset_name: Live map
        entity: vacuum.roborocks7
        map_source:
          camera: camera.xiaomi_cloud_map_extractor
        calibration_source:
          camera: true
        map_modes:
        - template: vacuum_clean_segment
          repeats_type: EXTERNAL
          max_repeats: 3
          service_call_schema:
            service: vacuum.send_command
            service_data:
              command: app_segment_clean
              entity_id: '[[entity_id]]'
              params:
                - segments: '[[selection]]'
                  repeat: '[[repeats]]'
          predefined_selections:
          - id: 16
            icon:
              name: mdi:sofa
              x: 24825
              'y': 22075
          - id: 17
            icon:
              name: mdi:chef-hat
              x: 20000
              'y': 23950
          - id: 18
            icon:
              name: mdi:bed
              x: 24950
              'y': 16500
          - id: 19
            icon:
              name: mdi:desk
              x: 20675
              'y': 21075
          - id: 20
            icon:
              name: mdi:food
              x: 21900
              y: 23500
        - template: vacuum_clean_zone_predefined
          predefined_selections:
          - zones: [[ 20928, 21938, 22500, 24887 ]]
            icon:
              name: "mdi:table"
              x: 21800
              y: 23428
          - zones: [[ 21107, 24693, 23649, 25831 ]]
            icon:
              name: "mdi:emoticon-poop-outline"
              x: 22500
              y: 25300
        - template: vacuum_clean_zone
        - template: vacuum_goto
      - type: custom:more-info-card
        entity: vacuum.roborocks7
               
```

**Comment by yaruslavm** [comment]
i have the same issue 
i submitted a new bug and only after reading this thread once again i saw that this issue was mentioned here.
#398
